nixos/modules/tasks/filesystems.nix

```diff
@@ -290,11 +299,10 @@ in
       "/dev" = { fsType = "devtmpfs"; options = [ "nosuid" "strictatime" "mode=755" "size=${config.boot.devSize}" ]; };
       "/dev/shm" = { fsType = "tmpfs"; options = [ "nosuid" "nodev" "strictatime" "mode=1777" "size=${config.boot.devShmSize}" ]; };
       "/dev/pts" = { fsType = "devpts"; options = [ "nosuid" "noexec" "mode=620" "gid=${toString config.ids.gids.tty}" ]; };
-    } // optionalAttrs (!config.boot.isContainer) {
       # systemd-nspawn populates /sys by itself, and remounting it causes all
       # kinds of weird issues (most noticeably, waiting for host disk device
       # nodes).
-      "/sys" = { fsType = "sysfs"; options = [ "nosuid" "noexec" "nodev" ]; };
+      "/sys" = { enabled = !config.boot.isContainer; fsType = "sysfs"; options = [ "nosuid" "noexec" "nodev" ]; };
     };
```

Comment:
@dezgeg @abbradar No, because then we have to encode too much logic in nixpkgs. For example, the `libvirt-lxc` container sets up `/dev/pts` itself and the console breaks if NixOS also mounts it. So we'd like to be able to turn `/dev/pts` off from an external, LXC-specific module.

Why do we need to support `enable` in regular `filesystems`? I can't find any place in nixpkgs where we do something like `filesystems = config.boot.specialFileSystems`. The regular file systems are different, because NixOS doesn't predefine a bunch of mountpoints in that case.

Btw, I started a general discussion of this "unsetting" of options here: http://lists.science.uu.nl/pipermail/nix-dev/2016-September/021663.html .